sql/internal/specutil/convert.go

```diff
@@ -34,49 +34,75 @@ type (
 )
 
 // Realm converts the schemas and tables into a schema.Realm.
-func Realm(schemas []*sqlspec.Schema, tables []*sqlspec.Table, convertTable ConvertTableFunc) (*schema.Realm, error) {
-	r := &schema.Realm{}
+func Realm(r *schema.Realm, schemas []*sqlspec.Schema, tables []*sqlspec.Table, convertTable ConvertTableFunc) error {
 	for _, schemaSpec := range schemas {
 		var schemaTables []*sqlspec.Table
 		for _, tableSpec := range tables {
 			name, err := SchemaName(tableSpec.Schema)
 			if err != nil {
-				return nil, fmt.Errorf("specutil: cannot extract schema name for table %q: %w", tableSpec.Name, err)
+				return fmt.Errorf("specutil: cannot extract schema name for table %q: %w", tableSpec.Name, err)
 			}
 			if name == schemaSpec.Name {
 				schemaTables = append(schemaTables, tableSpec)
 			}
 		}
-		sch, err := Schema(schemaSpec, schemaTables, convertTable)
+		sch := &schema.Schema{Name: schemaSpec.Name, Realm: r}
+		if err := Schema(sch, schemaTables, convertTable); err != nil {
+			return err
+		}
+		r.Schemas = append(r.Schemas, sch)
+	}
+	for _, sch := range r.Schemas {
+		for _, tbl := range sch.Tables {
+			tableSpec, err := findTableSpec(tables, sch.Name, tbl.Name)
+			if err != nil {
+				return err
+			}
+			if err := LinkForeignKeys(tbl, sch, tableSpec); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// findTableSpec searches tableSpecs for a spec of a table named tableName in a schema named schemaName.
+func findTableSpec(tableSpecs []*sqlspec.Table, schemaName, tableName string) (*sqlspec.Table, error) {
+	for _, tbl := range tableSpecs {
+		n, err := SchemaName(tbl.Schema)
 		if err != nil {
 			return nil, err
 		}
-		r.Schemas = append(r.Schemas, sch)
+		if n == schemaName && tbl.Name == tableName {
+			return tbl, nil
+		}
 	}
-	return r, nil
+	return nil, fmt.Errorf("table %s.%s not found", schemaName, tableName)
 }
 
 // Schema converts a sqlspec.Schema with its relevant []sqlspec.Tables
 // into a schema.Schema.
-func Schema(spec *sqlspec.Schema, tables []*sqlspec.Table, convertTable ConvertTableFunc) (*schema.Schema, error) {
-	sch := &schema.Schema{
-		Name: spec.Name,
-	}
+func Schema(sch *schema.Schema, tables []*sqlspec.Table, convertTable ConvertTableFunc) error {
 	m := make(map[*schema.Table]*sqlspec.Table)
 	for _, ts := range tables {
 		table, err := convertTable(ts, sch)
 		if err != nil {
-			return nil, err
+			return err
 		}
 		sch.Tables = append(sch.Tables, table)
 		m[table] = ts
 	}
+	// If this is part of a realm, linking will be done on the Realm building since FKs can exist
+	// between different schemas in the same realm.
```

Comment:
```suggestion
	// If this is part of a realm, linking will be done on the Realm building
	// since FKs can exist between different schemas in the same realm.
```